src/abstractdataframe/abstractdataframe.jl

```diff
@@ -645,79 +658,16 @@ julia> dropmissing(df, [:x, :y])
 ```
 
 """
-dropmissing(df::AbstractDataFrame,
-            cols::Union{Integer, Symbol, AbstractVector}=1:size(df, 2)) =
-    deleterows!(copy(df), findall(!, completecases(df, cols)))
-
-"""
-    dropmissing!(df::AbstractDataFrame)
-    dropmissing!(df::AbstractDataFrame, cols::AbstractVector)
-    dropmissing!(df::AbstractDataFrame, cols::Union{Integer, Symbol})
-
-Remove rows with missing values from data frame `df` and return it.
-If `cols` is provided, only missing values in the corresponding columns are considered.
-
-See also: [`dropmissing`](@ref) and [`completecases`](@ref).
-
-# Examples
-
-```jldoctest
-julia> df = DataFrame(i = 1:5,
-                      x = [missing, 4, missing, 2, 1],
-                      y = [missing, missing, "c", "d", "e"])
-5×3 DataFrame
-│ Row │ i     │ x       │ y       │
-│     │ Int64 │ Int64⍰  │ String⍰ │
-├─────┼───────┼─────────┼─────────┤
-│ 1   │ 1     │ missing │ missing │
-│ 2   │ 2     │ 4       │ missing │
-│ 3   │ 3     │ missing │ c       │
-│ 4   │ 4     │ 2       │ d       │
-│ 5   │ 5     │ 1       │ e       │
-
-julia> df1 = copy(df);
-
-julia> dropmissing!(df1);
-
-julia> df1
-2×3 DataFrame
-│ Row │ i     │ x      │ y       │
-│     │ Int64 │ Int64⍰ │ String⍰ │
-├─────┼───────┼────────┼─────────┤
-│ 1   │ 4     │ 2      │ d       │
-│ 2   │ 5     │ 1      │ e       │
-
-julia> df2 = copy(df);
-
-julia> dropmissing!(df2, :x);
-
-julia> df2
-3×3 DataFrame
-│ Row │ i     │ x      │ y       │
-│     │ Int64 │ Int64⍰ │ String⍰ │
-├─────┼───────┼────────┼─────────┤
-│ 1   │ 2     │ 4      │ missing │
-│ 2   │ 4     │ 2      │ d       │
-│ 3   │ 5     │ 1      │ e       │
+function dropmissing(df::AbstractDataFrame,
+                     cols::Union{Integer, Symbol, AbstractVector}=1:size(df, 2);
+                     keepeltype::Bool=true)
+    if keepeltype
+        Base.depwarn("dropmissing will change eltype of cols to disallow Missing by default. " *
```

Comment:
```suggestion
        Base.depwarn("dropmissing will change eltype of cols to disallow missing by default. " *
```
Should also give the syntax which allows getting the new behavior to turn off the warning.